hooks/image/70-flatpak-appstream-catalog

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+# Construct AppStream XML for preinstalled apps
+
+import logging
+import os
+import subprocess
+
+from xml.etree import ElementTree
+
+import gi
+
+gi.require_version("Flatpak", "1.0")
+from gi.repository import GLib, Gio, Flatpak  # noqa: E402
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="+ %(asctime)s %(levelname)s %(name)s: %(message)s",
+    datefmt="%H:%M:%S",
+)
+logger = logging.getLogger(os.path.basename(__file__))
+
+APPSTREAM_VERSION = "0.8"
+
+
+def main():
+    GLib.set_prgname(os.path.basename(__file__))
+
+    image_version = os.environ["EIB_OUTVERSION"]
+    appstream_filename = image_version + ".appstream.xml"
+    appstream_path = os.path.join(os.environ["EIB_OUTDIR"], appstream_filename)
+
+    catalog = ElementTree.Element(
+        "components", version=APPSTREAM_VERSION, origin=image_version
+    )
+
+    system_path = os.path.join(os.environ["OSTREE_VAR"], "lib/flatpak")
+    logger.info("Opening Flatpak installation in %s", system_path)
+    system_file = Gio.File.new_for_path(system_path)
+    system = Flatpak.Installation.new_for_path(system_file, user=False)
+
+    installed_refs_by_origin_arch = {}
+    for ref in system.list_installed_refs():
+        key = (ref.get_origin(), ref.get_arch())
+        installed_refs_by_origin_arch.setdefault(key, set()).add(ref.format_ref())
+
+    for (origin, arch), installed_refs in installed_refs_by_origin_arch.items():
+        path = os.path.join(
+            system_path, "appstream", origin, arch, "active", "appstream.xml"
+        )
+
+        try:
+            metainfo = ElementTree.parse(path).getroot()
+        except ElementTree.ParseError as e:
+            # TODO: Should this be fatal?
+            logger.warning("Failed to parse %s: %s", path, e)
+            continue
+
+        metainfo_version = metainfo.attrib["version"]
+        if metainfo_version != APPSTREAM_VERSION:
```

Comment:
Unfortunately, this seems likely to fail sooner than later - https://blog.tenstral.net/2023/11/appstream-1-0-released.html. I don't think you should try harder than this right now, but eventually it will rear it's head where some remotes are publishing 0.8 and others are publishing 1.0. Although, maybe `flatpak build-update-repo` will always enforce 0.8 when generating the catalog or something like that.

Reply:
I agree it's likely to fail and I don't really know what to do about it. Perhaps just delete this whole block and assume it's good enough for the time being; and revisit once Flathub makes the noisy transition to appstream 1.0?